### PR TITLE
adding multithreading to coordinator

### PR DESCRIPTION
**Pull request by jeremy23newman**

### CODE REVIEW

File: src/factorFinder/FactorComputationCoordinatorImpl.java [hunk 1]
```diff
@@ -1,39 +1,57 @@
...
-        List<String> formattedFactors =
-            computeEngine.getFormattedFactors(number);
-        FileOutputConfig outputConfig = request.getOutputConfig();
+        Callable<List<String>> task = () -> computeEngine.getFormattedFactors(number);
+        futures.add(executor.submit(task)); // Submit each task to the executor
+      }
+
+      // Process each future result
+      for (Future<List<String>> future : futures) {
+        List<String> formattedFactors = future.get(); // Retrieve computed factors
```
Comment (ViceRaven) [inline src/factorFinder/FactorComputationCoordinatorImpl.java:48]: I tested your code inside mine in eclipse. It seems to work well. The Gradle still passes for test multi user.

File: src/factorFinder/FactorComputationCoordinatorImpl.java
Comment (DijonnaWatson) [inline src/factorFinder/FactorComputationCoordinatorImpl.java:1]: Everything looks good to me, nothing's missing and TestMultiUser still passes in my eclipse. Good job Jeremy

File: src/factorFinder/FactorComputationCoordinatorImpl.java [hunk 2]
```diff
@@ -1,39 +1,57 @@
...
+      List<Future<List<String>>> futures = new ArrayList<>();
+      
+      // Passes the integers to the compute component
       for (Integer number : inputData) {
-        List<String> formattedFactors =
-            computeEngine.getFormattedFactors(number);
-        FileOutputConfig outputConfig = request.getOutputConfig();
+        Callable<List<String>> task = () -> computeEngine.getFormattedFactors(number);
+        futures.add(executor.submit(task)); // Submit each task to the executor
+      }
```
Comment (DijonnaWatson) [inline src/factorFinder/FactorComputationCoordinatorImpl.java:44]: Everything looks good to me, nothing's missing and TestMultiUser still passes in my eclipse. I see both of you guys did a little bit on 2, Good job on fixing it guys.

**Review by DijonnaWatson — COMMENTED** [review]
I think everything looks good, and since you have the TestMultiUser, TestUser, and testInputFile, I added, I think we can just merge your branch to main.

**Review by DijonnaWatson — APPROVED** [review]
I think everything looks good, and since you have the TestMultiUser, TestUser, and testInputFile, I added, I think we can just merge your branch to main.

File: src/factorFinder/FactorComputationCoordinatorImpl.java [hunk 3]
```diff
@@ -43,6 +47,6 @@ public FactorComputeResult compute(FactorComputeRequest request) {
       return FactorComputeResult.FAILURE;
     }
 
-    return FactorComputeResult.SUCCESS;
+    return FactorComputeResult.SUCCESS;//Both the input and output file paths were correct and existed
```
Comment (jeremy23newman) [inline src/factorFinder/FactorComputationCoordinatorImpl.java]: Nice comments, helps the readability and good job importing/troubleshooting the test classes

**Pull request merged**